src/main/java/neqsim/thermodynamicoperations/flashops/Flash.java

```diff
@@ -397,36 +397,35 @@ public void solidPhaseFlash() {
         tempVar[k] = system.getPhase(0).getComponent(k).getz();
         for (int i = 0; i < system.getNumberOfPhases() - 1; i++) {
           tempVar[k] -=
-              system.getBeta(i) * system.getPhases()[3].getComponent(k).getFugacityCoefficient()
+              system.getBeta(i) * system.getPhase(3).getComponent(k).getFugacityCoefficient()
```

Comment:
In this case I beleive we have to used system.getPhases()[3] - as the solid phase is always number three (and getPhase(3) can point to a different phase.